app/Libs/HtmlValidator/Validator.php

```diff
@@ -0,0 +1,308 @@
+	const CHARSET_TIS_620 = 'TIS-620';
+	const CHARSET_GBK = 'GBK';
+	const CHARSET_GB18030 = 'GB18030';
+	const CHARSET_BIG5 = 'Big5';
+	const CHARSET_BIG5_HKSCS = 'Big5-HKSCS';
+	const CHARSET_SHIFT_JIS = 'Shift_JIS';
+	const CHARSET_ISO_2022_JP = 'ISO-2022-JP';
+	const CHARSET_EUC_JP = 'EUC-JP';
+	const CHARSET_ISO_2022_KR = 'ISO-2022-KR';
+	const CHARSET_EUC_KR = 'EUC-KR' ;
+
+	/**
+	 * Default validator URL
+	 *
+	 * @var string
+	 */
+	const DEFAULT_VALIDATOR_URL = 'https://html5.validator.nu';
+
+	/**
+	 * Holds the HTTP client used to communicate with the API
+	 *
+	 * @var GuzzleHttp\Client
+	 */
+	private $httpClient;
+
+	/**
+	 * Parser to use for validating
+	 *
+	 * @var string
+	 */
+	private $parser;
+
+	/**
+	 * Default charset to report to the validator
+	 *
+	 * @var string
+	 */
+	private $defaultCharset = self::CHARSET_UTF_8;
+
+	/**
+	 * Node wrapper tool
+	 *
+	 * @var HtmlValidator\NodeWrapper
+	 */
+	private $nodeWrapper;
+
+	/**
+	 * Constructs a new validator instance
+	 */
+	public function __construct($validatorUrl = self::DEFAULT_VALIDATOR_URL, $parser = self::PARSER_HTML5) {
+		$this->httpClient = new HttpClient([
+			'base_uri' => $validatorUrl,
+			'headers' => ['User-Agent' => 'rexxars/html-validator']
+		]);
+
+		$this->nodeWrapper = new NodeWrapper();
+
+		$this->setParser($parser);
+	}
+
+	/**
+	 * Set the HTTP client to use for requests
+	 *
+	 * @param GuzzleHttp\ClientInterface $httpClient
+	 * @return HttpValidator\Validator
+	 */
+	public function setHttpClient($httpClient) {
+		$this->httpClient = $httpClient;
+
+		return $this;
+	}
+
+	/**
+	 * Get the set parser type
+	 *
+	 * @return string
+	 */
+	public function getParser() {
+		return $this->parser;
+	}
+
+	/**
+	 * Set parser to use for the given markup
+	 *
+	 * @param string $parser Parser name (use Validator::PARSER_*)
+	 * @return Validator Returns current instance
+	 * @throws UnknownParserException If parser specified is not known
+	 */
+	public function setParser($parser) {
+		switch ($parser) {
+			case self::PARSER_XML:
+			case self::PARSER_XMLDTD:
+			case self::PARSER_HTML:
+			case self::PARSER_HTML5:
+			case self::PARSER_HTML4:
+			case self::PARSER_HTML4TR:
+				$this->parser = $parser;
+				return $this;
+			default:
+				throw new UnknownParserException('Unknown parser "' . $parser . '"');
+		}
+	}
+
+	/**
+	 * Get the charset to report to the validator
+	 *
+	 * @return string
+	 */
+	public function getCharset() {
+		return $this->defaultCharset;
+	}
+
+	/**
+	 * Set the charset to report to the validator
+	 *
+	 * @param string $charset Charset name (defaults to 'utf-8')
+	 * @return HtmlValidator\Validator
+	 */
+	public function setCharset($charset) {
+		$this->defaultCharset = $charset;
+
+		return $this;
+	}
+
+	/**
+	 * Get the correct mime-type for the given parser
+	 *
+	 * @param  string $parser Parser name
+	 * @return string
+	 */
+	private function getMimeTypeForParser($parser) {
+		switch ($parser) {
```

Comment:
![CRITICAL](https://sonarsource.github.io/sonar-github/severity-critical.png 'Severity: CRITICAL') Add a "case default" clause to this "switch" statement. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](http://sonar.relipa.blog/coding_rules#rule_key=php%3AS131)